File: 038 Count and Say/count_and_say.py
class Solution: # Author: XC
    # @param {integer} n
    # @return {string}
    def countAndSay(self, n):
        s = '1'
        result = []
        for i in range(n - 1):
            count = 1
            current = s[0]
            for j in range(1, len(s)):
                if s[j] == current:
                    count += 1
                else:
                    result.append(str(count))
                    result.append(current)
                    current = s[j]
                    count = 1
            result.append(str(count))
            result.append(current)
            s = ''.join(result)
            result = []
        return s

File: 038 Count and Say/count_and_say.cpp
class Solution {
public:
    string countAndSay(int n) {
        string s = "1";
        stringstream ss;
        char c;
        int count;
        for (int i = 1; i < n; i++) {
            ss.str(string());
            c = s[0];
            count = 1;
            for (int j = 1; j < s.size(); j++) {
                if (s[j] == c) count++;
                else {
                    ss << count << c;
                    c = s[j];
                    count = 1;
                }
            }
            ss << count << c;
            s = ss.str();
        }
        return s;
    }
}; // Author: XC